src/System.Data.SqlClient/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -1071,5 +1071,73 @@ static internal Version GetAssemblyVersion()
 
             return s_systemDataVersion;
         }
+
+
+        static internal readonly string[] AzureSqlServerEndpoints = {Res.GetString(Res.AZURESQL_GenericEndpoint),
+                                                                    Res.GetString(Res.AZURESQL_GermanEndpoint),
+                                                                    Res.GetString(Res.AZURESQL_UsGovEndpoint),
+                                                                    Res.GetString(Res.AZURESQL_ChinaEndpoint) };
+
+        static internal bool IsAzureSqlServerEndpoint(string dataSource)
+        {
+            Debug.Assert(!string.IsNullOrWhiteSpace(dataSource));
+
+            string serverName = ExtractServerName(dataSource);
```

Comment:
With ExtractServerName() and without ExtractServerName(), there is only 2 lines change.  '''serverName = serverName.Replace("np:", "");''' and '''serverName = serverName.Replace("tcp:", "");''' The rest of them is all needed.